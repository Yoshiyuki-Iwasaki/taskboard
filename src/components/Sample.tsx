import { useRouter } from "next/router";
import React from "react";

export default function Sample() {
  const router = useRouter();
  const path = router.asPath;

  return (
    <>
      <h1>このページの path は {path} です。</h1>
      <input data-testid="input" value="" type="text" onChange={()=>console.log('test')}/>
    </>
  );
}
