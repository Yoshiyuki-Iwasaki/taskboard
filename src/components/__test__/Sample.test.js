import { render } from "@testing-library/react";
import React from "react";
import Sample from "../Sample";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      asPath: "/",
    };
  },
}));

describe("初期表示の確認", () => {
  test("サンプルページ", () => {
    const { asFragment } = render(<Sample />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the correct initial DOM", () => {
    const doc = render(<Sample />);
    const inputElement = doc.getByTestId("input");
    // The input should be blank.
    expect(inputElement.getAttribute("value")).toBe("");
  });
});