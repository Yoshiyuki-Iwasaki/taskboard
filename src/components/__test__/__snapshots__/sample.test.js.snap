// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`初期表示の確認 サンプルページ 1`] = `
<DocumentFragment>
  <h1>
    このページの path は / です。
  </h1>
  <input
    data-testid="input"
    type="text"
    value=""
  />
</DocumentFragment>
`;
